### ISSUE TITLE
Can't copy RasterMonochrome

### ISSUE THREAD

**Issue by davidpiuva** [cogneco/ooc-kean#500]
"copy" in RasterMonochrome crashes every time in ooc-kean/source/base/ByteBuffer.ooc.

**Comment by marcusnaslund** [comment]
Looks like `_buffer` in `RasterPacked` isn't always initialized. Change the `~fromOriginal` constructor in `RasterPacked` to:

```ooc
init: func ~fromOriginal (original: This) {
	super(original)
	this _buffer = ByteBuffer new(original _buffer size)
	this _stride = original stride
}
```

**Comment by sebastianbaginski** [comment]
Could be nice to add test cases too :)

**Comment by marcusnaslund** [comment]
Oh definitely. The following methods are not covered at all in the current `RasterMonochrome` test:
* create
* copy
* apply
* getFirstDerivativeWindowOptimized
* getFirstDerivativeWindow
* getValue
* getRowInto
* getColumnInto

**Comment by sebastianbaginski** [comment]
The last two are called by `getRow` and `getColumn` internally.
`apply` is called internally by `convertFrom`.

**Comment by davidpiuva** [comment]
RasterPacked's ~fromOriginal is not even called for some weird reason. It skips directly to RasterImage

**Comment by marcusnaslund** [comment]
Yes, but `~fromRasterImage` only calls `super(original)`, and the appropriate super constructor is `~fromOriginal` in `RasterPacked`. This (appears to have) solved the problem for me, anyhow.

**Comment by davidpiuva** [comment]
Do you want to take over the issue? ~fromRasterImage won't call ~fromOriginal for me.

**Comment by marcusnaslund** [comment]
Sure!

**Comment by marcusnaslund** [comment]
I see what you mean now (and don't understand how I did this yesterday). Because the argument to `~fromRasterImage` is of type `RasterImage`, super for some reason is not the super of `RasterMonochrome` but the super of `RasterImage`. I solved this by creating the buffer directly in `copy`.
I am not too knowledgable about how these classes are supposed to work, so can't really write all these tests. Please look into that when you have time.

**Comment by marcusnaslund** [comment]
Solved by #504.
